### PR TITLE
Enable Tropical Stay Example App in Gitpod

### PR DESCRIPTION
**Pull request by GreenBusDriver**
## Change Description
Did the necessary setup to run Tropical Stay in Gitpod

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  onOpen: ignore
+- port: 8888
+  onOpen: ignore
+- port: 9876
+  onOpen: ignore
+
+tasks:
+- name: WebApp
+  init: yarn
+  # We need the $PWD here because this path is not resolved from the current folder but from the path nginx runs from.  $PWD is the workspace path where the copy of the repo files is
```
Comment (TaraTritt) [inline .gitpod.yml]: Nitpicky but `$PWD is the workspace path where the copy of the repo files` **is** -> `$PWD is the workspace path where the copy of the repo files` **are**

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  prebuilds:
+    branches: true
+
+ports:
+- port: 3000
+  onOpen: ignore
+# We open this port because it's the port that nginx is serving and which is federating the other services (web app and chain in this case)
+- port: 8000
+  onOpen: open-preview
+- port: 8900
```
Comment (TaraTritt) [inline .gitpod.yml]: Are all these ports necessary?

Comment (TaraTritt) [reply .gitpod.yml]: Obviously 3000 for the app, 8000 for nginx, and 8888 for nodeos not familiar with the rest

File: eosio/scripts/deploy_contracts.sh [hunk 3]
```diff
@@ -175,15 +224,28 @@ transfer_sys_tokens example
 # eosio.assert actions
 # Set chain
 assert_set_chain "cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f" "Local Chain" "8ae3ccb19f3a89a8ea21f6c5e18bd2bc8f00c379411a2d9319985dad2db6243e"
-# Register tropical manifest
-assert_register_manifest "tropical" "http://localhost:3000" "http://localhost:3000/app-metadata.json#bc677523fca562e307343296e49596e25cb14aac6b112a9428a42119da9f65fa" "[{ "\""contract"\"": "\""tropical"\"",  "\""action"\"": "\""like"\"" }]"
 
+# Register tropical manifest
+# If running in GitPos, we need to alter the URLs
```
Comment (TaraTritt) [inline eosio/scripts/deploy_contracts.sh]: `Gitpos` -> `Gitpod`

File: README.md [hunk 4]
```diff
@@ -5,10 +5,27 @@ Tropical Example is a mock application for renting properties. It will be refere
...
+
+Gitpod [launches the app](https://gitpod.io/#https://github.com/EOSIO/tropical-example-web-app) for you. It starts the required blockchain in the background, launches the web server, and opens a preview window.
+NOTES: 
+ 1) There are several times during startup it might look like startup hangs, namely... near the end of the docker build, once the IDE comes up, and then once the preview shows.
+ 2) Sometimes when Gitpod launches the webapp preview, it does so prematurely.  Just click the small refresh circular arrow icon in the top left of the preview window.
+ 3) Gitpod generates a dynamic URL for the browser to access the app from. This URL is needed in numerous parts of the app, so note that there is code in this repo specifically meant for Gitpod compatibility. A comment has been added in those locations to point it out.
+ 4) To use Scatter in Gitpod, launch the demo in Gitpod, then go into Scatter's Networks section and add a custom network with the following settings as well as adding the account name. Note that these settings will need to be updated each time you launch the demo in Gitpod (because the URL will be different each time).
+ -Network Settings
```
Comment (TaraTritt) [inline README.md]: The formatting of `-Network Settings` and `-Adding account name` is a little wonky when the README is rendered:

Screen Shot 2019-11-01 at 1 38 30 PM

Comment (TaraTritt) [reply README.md]: Might also help to bold `Host`, `Protocol` etc just to make everything easier to read

File: README.md [hunk 5]
```diff
@@ -5,10 +5,27 @@ Tropical Example is a mock application for renting properties. It will be refere
...
+  - Host: <the hostname (URL *without* the "https://" or a final "/") Gitpod generated to view the web page; it's what's in the address bar of the browser it opened for you>
+  - Protocol: https
+  - Port: 443
+  - Chain ID: cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f (This is the test chainId used in the example)
+ -Adding account name
+  - After you've imported the private key from this example (see other parts of the README for those instructions), Scatter might not pull the "example" account from the network. If that's the case, in the Wallet section, if "example" doesn't show up under your imported key with a balance to the right, on a Mac, you'll hold Ctrl when you click the three horizontal dot button to the right of your imported key. Ctrl will enable a normally-hidden item called "Link Account". Click that and in the first box, type "example" and in the drop-down, select the custom network you created above.
```
Comment (TaraTritt) [inline README.md]: Not sure if it's worth it, but this is a bit hard to follow. Might be worth adding pictures?

File: .gitpod.dockerfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+USER root
+
+ENV ROOT_DIR="/home/gitpod"
+RUN echo "INSTALLING EOSIO AND CDT" \
+ && apt-get update -y \
+ && apt-get install -y wget sudo curl \
+ && wget https://github.com/EOSIO/eosio.cdt/releases/download/v1.6.1/eosio.cdt_1.6.1-1_amd64.deb \
+ && apt-get update && sudo apt install -y --allow-downgrades ./eosio.cdt_1.6.1-1_amd64.deb \
+ && wget https://github.com/EOSIO/eos/releases/download/v1.7.3/eosio_1.7.3-1-ubuntu-18.04_amd64.deb \
```
Comment (tbfleming) [inline .gitpod.dockerfile]: Not required now. It'd be nice to update Tropical Stay to not need these downgrades.

File: .gitpod.yml (hunk 2 above)
Comment (GreenBusDriver) [reply .gitpod.yml]: Yes, any port not listed here (that is actually being used) will spam the user with a little pop-up box, so I added 8900 (keosd) to stop that.  All these ports are used by the system.
